core/FrontController.php

```diff
@@ -743,4 +748,30 @@ private static function setRequestIdHeader()
         $requestId = self::getUniqueRequestId();
         Common::sendHeader("X-Matomo-Request-Id: $requestId");
     }
+
+    private function isSupportedBrowserCheckNeeded()
+    {
+        if (Piwik::getModule() === 'API' && (empty(Piwik::getAction()) || Piwik::getAction() === 'index' || Piwik::getAction() === 'glossary')) {
```

Comment:
btw one more thing...  we could technically return also `false` if `Common::isPhpCliMode()===true` meaning Matomo is used on CLI.

Indirectly this would be covered by the `is user agent empty check` below but could be also added additionally. Basically, on CLI there is no browser and thus no need to execute anything. The problem you will then run into is that unit tests run on CLI and thus it won't be executed there anymore. You would then need to add yet another check for `defined('PIWIK_TEST_MODE') && PIWIK_TEST_MODE` to detect if it's running in tests and then ignore this `is cli mode` check :) So maybe we rather just not check for CLI and rely on the `is user agent empty` check for now :)

Reply:
I'm happy to add the test mode check. this way it's clear what and why is happening